internal/services/fluidrelay/fluid_relay_servers_resource.go

```diff
@@ -312,3 +356,51 @@ func (s Server) Delete() sdk.ResourceFunc {
 func (s Server) IDValidationFunc() pluginsdk.SchemaValidateFunc {
 	return fluidrelayservers.ValidateFluidRelayServerID
 }
+
+func expandFluidRelayServerCustomerManagedKey(input []CustomerManagedKey) *fluidrelayservers.EncryptionProperties {
+	if len(input) == 0 {
+		return nil
+	}
+
+	v := input[0]
+	encryption := &fluidrelayservers.EncryptionProperties{
+		CustomerManagedKeyEncryption: &fluidrelayservers.CustomerManagedKeyEncryptionProperties{
+			KeyEncryptionKeyURL: pointer.To(v.KeyVaultKeyID),
+			KeyEncryptionKeyIdentity: &fluidrelayservers.CustomerManagedKeyEncryptionPropertiesKeyEncryptionKeyIdentity{
+				IdentityType:                   pointer.To(fluidrelayservers.CmkIdentityTypeUserAssigned),
+				UserAssignedIdentityResourceId: pointer.To(v.UserAssignedIdentityId),
+			},
+		},
+	}
+
+	return encryption
+}
+
+func flattenFluidRelayServerCustomerManagedKey(input *fluidrelayservers.EncryptionProperties) ([]CustomerManagedKey, error) {
+	if input == nil || input.CustomerManagedKeyEncryption == nil {
+		return []CustomerManagedKey{}, nil
+	}
+
+	customerManagedKey := CustomerManagedKey{}
+
+	if input.CustomerManagedKeyEncryption.KeyEncryptionKeyURL != nil {
+		if v := pointer.From(input.CustomerManagedKeyEncryption.KeyEncryptionKeyURL); v != "" {
+			id, err := commonids.ParseKeyVaultID(v)
+			if err != nil {
+				return []CustomerManagedKey{}, err
```

Comment:
Since we're error'ing here we should return `nil` instead of data that is valid
```suggestion
				return nil, err
```

Reply:
Done!